### PR TITLE
Moved game logic into a separate class

### LINKED ISSUE #36: Encapsulate game logic found in index.js into seperate classes / functions
#35 is an example of why this is useful / needed

### PR DESCRIPTION
**Pull request by ctcuff**
This PR takes the main game logic out of `index.js` and moves it into a new class `GameApp`. This PR also introduces a few changes:

* The manager now holds a `Camera` object that can be referenced via `manager.camera`. This allows the UFO to handle camera movement while still allowing `GameApp` to access camera uniforms and other camera methods
* A `gameEventEmitter` has been added. This is an `EventEmitter` that can broadcast certain game events. This helps restart the game when the ship runs out of lives (an example is in `index.js`)

Fixes #36

### CODE REVIEW

File: src/gameobjects/ufo.js [hunk 1]
```diff
@@ -13,7 +15,8 @@ class UFO extends GameObject {
     this.startRot = -300;
     this.physics = new Physics(new Vector3(0, 0, this.startSpeed), new Vector3(0, this.startRot, 0), 0);
     this.scale = 0.5;
-    this.initalDia = 99.7135610685365;
+    this.extents = manager.modelList.ufo.extents;
+    this.initalDia = this.extents.dia;
```
Comment (Armintron) [inline src/gameobjects/ufo.js:19]: Isnt this redundant?

**Review by Armintron — APPROVED** [review]
Other than a small comment, looks good to me!

**Comment by Armintron** [comment]
You also might want to merge #47 into this branch since it will conflict with `index.js`

File: src/gameobjects/ufo.js (hunk 1 above)
Comment (ctcuff) [reply src/gameobjects/ufo.js:19]: Actually, `this.initalDia` isn't even changed, only referenced, so I can just rename it to `dia`

**Comment by ctcuff** [comment]
> You also might want to merge #47 into this branch since it will conflict with `index.js`

Yep, wanna get that merged in first

**Comment by Armintron** [comment]
This PR also moves UFO input logic into the `UFO` class

**Pull request merged**